README.md
# DISLOCK v2.3

DISLOCK is the most advanced Discord Browser Generator.

It is capable of generating Unlocked Tokens for free by Using AI.
DISLOCK is currently undetected by Discord because its Human Emulation.

You will have to use HQ Proxies/IPs to get unlocked tokens.

## Features

- TokenGenerator on discord.com [Almost always Unlocked]
- TokenGenerator on discord.com/register [Mostly Unlocked]
- Captcha Tester on hcaptcha.com

## Demo Videos

### Unclaimed Generator
https://streamable.com/4wvhdw

### Normal Generator
https://streamable.com/w9l2fz

## Proxies

You will have to use HQ Proxies/IPs to get Unlocked Tokens.
If you really want to generate proxies, you maight have to spend fairly big amounts of money, to get undetected/unflagged IPs.
If you just want to test the Generator, you can also just restart your InternetRouter (if you have a rotating IP) and Discord wont notice.

(Btw your Proxy AD can stand here, DM me for offers ;d ;:D)

## Botright

This bot uses Botright, my Browser Automation Package. You can check it out [here](https://github.com/Vinyzu/Botright).

I would appreciate a star on this project aswell and hope that youll have fun creating your own bots with it!


## Installation

### Installing DISLOCK with Python

```bash
  git clone https://github.com/Vinyzu/DiscordGenerator DISLOCK
  cd DISLOCK
  pip install -r requirements.txt
  playwright install
  python main.py
```

### Further Requirements

- Windows

- [Git](https://git-scm.com/downloads) (To install DISLOCK)
- [Pip](https://pip.pypa.io/en/stable/installation/) (To install DISLOCK)

## Using

Type     | Recommended Usage              |
 :------- | :------------------------- |
| `Token Generator` | Generating HQ token (sometimes) locked the classic way |
| `Unclaimed Generator` | Generating Unclaimed, HQ (mostly) unlocked tokens  |
| `Captcha Tester` | Testing the CaptchaAI on hCaptcha.com |

##### Usages will be updated when Discord fixxes Modes

## Contributing

Contributions are always welcome!

See [Contributing](https://github.com/Vinyzu/DiscordGenerator/blob/main/contributing.md) for ways to get started.


## To the Skids

Hello, skid. I know its in your nature to laboriously copy and paste this project and sell it as yours. And i can´t 100% prevent that. However, legally you aren´t allowed to share your skidded DISLOCK other than the source code. And i know that you give a fuck about Licenses and Copyright, but if you gonna use this code as yours and don´t mark me as the original author, i can assure you that you won´t have a good time selling this ;d.

## Copyright and License
© [Vinyzu](https://github.com/Vinyzu/)

[GNU GPL](https://choosealicense.com/licenses/gpl-3.0/)

(Commercial Usage is allowed, but source, license and copyright has to made available. DISLOCK does not provide and Liability or Warranty)

## Authors

- [@Vinyzu](https://github.com/Vinyzu)

`If you appreciate this Repository, I would love to see you star and share this. It took a lot of effort and time to code all of those features and i originally planned to sell this project, so I´m "wasting" money for everyone´s fun ;:D.`



[![mjolnir-discord](https://img.shields.io/badge/Mjolnir_Discord-000?style=for-the-badge&logo=discord&logoColor=white)](https://discord.gg/rpd4gzrqGN)
[![my-discord](https://img.shields.io/badge/My_Discord-000?style=for-the-badge&logo=google-chat&logoColor=blue)](https://discordapp.com/users/935224495126487150)
[![buy-me-a-coffee](https://img.shields.io/badge/Buy_Me_A_Coffee-000?style=for-the-badge&logo=ko-fi&logoColor=brown)](https://ko-fi.com/vinyzu)


## Thanks to

[QIN2DIM](https://github.com/QIN2DIM/) (For his great AI work.)

[MaxAndolini](https://github.com/MaxAndolini) (For shared knowledge of hCaptcha bypassing)

[Dönerbäcker](https://github.com/DoenerBaecker) (For Proxies)


![Version](https://img.shields.io/badge/DISLOCK-v2.3-blue)
![License](https://img.shields.io/badge/License-GNU%20GPL-green)
![Python](https://img.shields.io/badge/Python-v3.x-lightgrey)
![Platforms](https://img.shields.io/badge/Platform-win--32%20%7C%20win--64-lightgrey)


changelog.md
# Changelog

Every new change will be logged

## v2.3
```
+ Closing PopUps after Generation
+ Better ErrorHandling
```

## v2.2
```
+ Fixxed Bug causing /register Gen not clicking Tos-Checkbox
+ Updated Email Claiming to be manual (not request)
```

## v2.1
```
+ Changed Checkbox Locator
+ Uodated LockCheck to work now
+ More LockChecking in EmailVerifier
+ More Sleeping whilst verifying Mail
```

## v2.0
```
+ Using Botright now
+ Updated Joiner and Humanization to be undetected
+ More small changes
```

## v1.0.6
```
+ Updated AI to support newest Tasks
+ Updated Avatar Function (should work now ig lol)
+ New IP-API (hopefully will fix ProxyIssues)
+ Added ServerJoiner (Wasnt tested enough to say its 100% stable)

+ Updated InviteLink in ReadMe

```

## v1.0.5
```
+ Fixxed To Low Password Security
+ Fixxed self.person error
+ Changed IP Provider (Hopefully Fixxes Proxy Errors)
+ Rewrote ProxyCheck Invalid Error

```

## v1.0.4
```
+ Tokens will now be saved even when the gen crashes
+ OutputFormat can now be set
+ Passwords now dont have ":" in them anymore (Messed up output format)
+ Delete verify=False in requests (caused warnings)
+ Fixxed Issue that NormalGenerator would save tempmail even when EmailVerification was disabled.

```

## v1.0.3
```
+ Raised Timeouts to allow slow proxies to work properly
+ Changed usage of TempmailAPI from package to local file
+ Updated AI to check images three times for any valid images
+ Replaced Httpx with Requests

```

## v1.0.2
```
+ Updated Model Downloader (httpx doesnt work)
--> Now, Models dont have to be given in the repo
+ Updated Proxy Class (Splitter) and general proxy-object handling

```

## v1.0.1
```
+ Updated various Proxy Handlers
+ Updated MailVerifier to wait until the new token is received
+ Updated smooth_out_mouse() to fix https://github.com/Vinyzu/DiscordGenerator/issues/3 by adding an extra point
+ Updated DISCUM Client to reduce code amount and allow proxies to work

```

## v1.0
```
+ First Commit
```

mailinfo.py
import requests
import json
import random
import string

class MailInfo:
    global DOMAINS
    global DOMAIN
    global TOKEN
    
    with open("config.json", "r") as conf: config = json.load(conf)
    DOMAINS = config["EMAIL STUFF"]["CUSTOM DOMAINS"]
    DOMAIN = random.choice(DOMAINS)
    TOKEN = "qgk2auEHq94bq5w8s_Fk78Iyt3xJrs9U7ESND5SHWbo"

    def generateInbox(rush=False):
        CHARS = string.ascii_letters + string.digits
        ALIAS = "".join(random.choice(CHARS) for _ in range(11))
        EMAIL = f"{ALIAS}@{DOMAIN}"
        return Inbox(EMAIL, TOKEN)

    """
    getEmail gets the emails from an inbox object
    and returns a list of Email objects
    """
    
    def getEmails(inbox):
        s = TempMail.makeHTTPRequest(f"/custom/{TOKEN}/{DOMAIN}")
        data = json.loads(s)

        # if no emails are found, return an empty list
        # else return a list of email
        if data["email"] == None:
            return ["None"]
        else:
            emails = []
            for email in data["email"]:
                emails.append(Email(email["from"], email["to"], email["subject"], email["body"], email["html"], email["date"]))
            return emails


class Email:
    def __init__(self, sender, recipient, subject, body, html, date):
        # make the propertys immutable using @property
        self._sender = sender
        self._recipient = recipient
        self._subject = subject
        self._body = body
        self._html = html
        self._date = date

    @property
    def sender(self):
        return self._sender

    @property
    def recipient(self):
        return self._recipient

    @property
    def subject(self):
        return self._subject

    @property
    def body(self):
        return self._body

    @property
    def html(self):
        return self._html

    @property
    def date(self):
        return self._date

    def __repr__(self):
        return ("Email (sender={}, recipient={}, subject={}, body={}, html={}, date={} )".format(self.sender, self.recipient, self.subject, self.body, self.html, self.date))


class Inbox:
    def __init__(self, address, token):
        # make the propertys immutable using @property
        self._address = address
        self._token = token

    @property
    def address(self):
        return self._address

    @property
    def token(self):
        return self._token

    def __repr__(self):
        return ("Inbox (address={}, token={} )".format(self.address, self.token))


modules/discord.py
import random
import base64
import platform
import re
import tempfile
import os

import httpx

from modules import tempmail

res = httpx.get("https://discord.com/login").text
file_with_build_num = 'https://discord.com/assets/'+re.compile(r'assets/+([a-z0-9]+)\.js').findall(res)[-2]+'.js'
req_file_build = httpx.get(file_with_build_num).text
index_of_build_num = req_file_build.find('buildNumber')+24
DISCORD_BUILD_NUM = int(req_file_build[index_of_build_num:index_of_build_num+6])

class Discord:
    async def get_headers(self, payload):
        cookies = await self.browser.cookies()
        __dcfduid = [item for item in cookies if item['name'] == "__dcfduid"][0]["value"]
        __sdcfduid = [item for item in cookies if item['name'] == "__sdcfduid"][0]["value"]
        cookies = f"__dcfduid={__dcfduid}; __sdcfduid={__sdcfduid}"

        super_props = {"os": platform.system(), "browser":"Firefox", "release_channel":"stable", "client_version": self.browser.browser.version, "os_version": str(platform.version()), "os_arch": "x64" if platform.machine().endswith('64') else "x86", "system_locale": self.browser.faker.locale, "client_build_number": DISCORD_BUILD_NUM, "client_event_source": None}
        super_props = base64.b64encode(str(super_props).encode()).decode()

        headers = {
            "accept": "*/*",
            "accept-encoding": "gzip, deflate, br",
            "accept-language": "de,de-DE;q=0.9",
            "authorization": self.token,
            "content-length": str(len(str(payload))),
            "content-type": "application/json",
            "cookie": cookies,
            "origin": "https://discord.com",
            "referer": "https://discord.com/channels/@me",
            "sec-fetch-dest": "empty",
            "sec-fetch-mode": "cors",
            "sec-fetch-site": "same-origin",
            "user-agent": self.browser.faker.useragent,
            "x-discord-locale": "en",
            "x-super-properties": super_props,
        }
        return headers

    async def humanize_token(self):
        await self.page.goto("https://discord.com/channels/@me")
        await self.page.wait_for_timeout(1000)
        # Clicking Settings Button
        settings_button = self.page.locator('[class *= "button-12Fmur"]').last
        await settings_button.click()
        # Click Profile Button
        await self.page.wait_for_timeout(500)
        profile_button = self.page.locator('[class *= "item-3XjbnG"]').nth(6)
        await profile_button.click()

        await self.page.wait_for_timeout(random.randint(2000, 3000))

        # Setting Random Avatar
        pics = httpx.get("https://api.github.com/repos/itschasa/Discord-Scraped/git/trees/cbd70ab66ea1099d31d333ab75e3682fd2a80cff")
        random_pic = random.choice(pics.json().get("tree")).get("path")
        pic_url = f"https://raw.githubusercontent.com/itschasa/Discord-Scraped/main/avatars/{random_pic}"
        pic = httpx.get(pic_url).content

        temp_file = tempfile.NamedTemporaryFile(suffix='.jpg', delete=False)
        temp_file.write(pic)
        temp_file.file.seek(0)  # change the position to the beginning of the file

        self.page.on("filechooser", lambda file_chooser: file_chooser.set_files(temp_file.name))

        upload_avatar_button = self.page.locator('[class *= "buttonsContainer-12kYno"]').locator('[class *= "lookFilled-yCfaCM "]')
        await upload_avatar_button.click()

        await self.page.wait_for_timeout(random.randint(500, 1000))

        upload_own = self.page.locator('[class *= "file-input"]')
        await upload_own.click()

        await self.page.wait_for_timeout(random.randint(500, 1000))
        confirm_button = self.page.locator('[class *= "button-f2h6uQ"]').last
        await confirm_button.click()

        temp_file.close()
        os.unlink(temp_file.name)


        await self.page.wait_for_timeout(random.randint(2000, 3000))

        # Setting AboutME
        try:
            quote = httpx.get("https://free-quotes-api.herokuapp.com")
            quote = quote.json().get("quote")
        except:
            self.logger.warning('Couldnt Get a Random Quote, Setting "Dislock" as AboutMe')
            quote = "Dislock"

        profile_button = self.page.locator('[role="textbox"]')
        await profile_button.click()
        await self.page.keyboard.type(quote)

        await self.page.wait_for_timeout(random.randint(500, 1000))
        confirm_button = self.page.locator('[class *= "colorGreen-3y-Z79"]').last
        await confirm_button.click()

        # Going to Hypesquad Page
        hypesquad_button = self.page.locator('[aria-controls="hypesquad-online-tab"]')
        await hypesquad_button.click()

        await self.page.wait_for_timeout(random.randint(5000, 6000))

        # Setting Hypesquad
        # payload = {"house_id": random.randint(1, 3)}
        # headers = await Discord.get_headers(self, payload)
        # hypesquad = await self.page.request.post("https://discord.com/api/v9/hypesquad/online", data=payload, headers=headers)
        # print(await hypesquad.json())

        is_locked = await Discord.is_locked(self)
        if is_locked:
            self.logger.error(f"Token {self.token} got locked whilst humanizing!")
            await self.close()
            return

        self.log_output()
        self.logger.info(f"Set Bio and ProfilePic!")

    async def join_server(self):
        await self.page.goto("https://discord.com/channels/@me")
        await self.page.wait_for_timeout(1000)
        # Clicking Join Server Button
        create_join_button = self.page.locator('[data-list-item-id *= "create-join-button"]')
        await create_join_button.click()
        await self.page.wait_for_timeout(500)
        # Clicking Join a Server Button
        another_server_button = self.page.locator('[class *= "footerButton-24QPis"]')
        await another_server_button.click()
        # Type Invite Code
        await self.page.wait_for_timeout(1000)
        await self.page.type("[placeholder='https://discord.gg/hTKzmak']", self.invite_link)
        # Clicking Join Server Button
        join_server_button = self.page.locator('[class *= "lookFilled-yCfaCM"]').last
        await join_server_button.click()

        try:
            await self.page.solve_hcaptcha()
        except:
            self.logger.info("No JoinServer Captcha detected.")

        is_locked = await Discord.is_locked(self)
        if is_locked:
            self.logger.error(f"Token {self.token} got locked whilst joining a Server!")
            await self.close()
            return

        self.log_output()
        self.logger.info("Joined Server successfully.")

    async def is_locked(self):
        token_check = await self.page.request.get('https://discord.com/api/v9/users/@me/library', headers={"Authorization": self.token})
        token_check = token_check.status == 200
        if token_check:
            r = await self.page.request.get('https://discord.com/api/v9/users/@me', headers={"Authorization": self.token})
            response = await r.json()
            self.id = response.get("id")
            self.email = response.get("email")
            self.username = response.get("username")
            self.discriminator = response.get("discriminator")
            self.tag = f"{self.username}#{self.discriminator}"
            self.flags = response.get("public_flags")

        return not token_check

    async def set_email(self, email):
        try:
            # Setting Email
            await self.page.goto("https://discord.com/channels/@me")
            await self.page.wait_for_timeout(1000)
            # Clicking Settings Button
            settings_button = self.page.locator('[class *= "button-12Fmur"]').last
            await settings_button.click()
            # Click Email Button
            await self.page.wait_for_timeout(500)
            settings_button = self.page.locator('[class *= "fieldButton-14lHvK"]').nth(1)
            await settings_button.click()

            # Typing Mail
            mail_input = self.page.locator('[type="text"]').last
            await mail_input.type(email)

            # Typing Password
            password_input = self.page.locator('[type="password"]').last
            await password_input.type(self.browser.faker.password)

            # Click Claim Button
            claim_button = self.page.locator('[type="submit"]').last
            await claim_button.click()
        except Exception as e:
            print(e)

    async def confirm_email(self):
        before_token = self.token
        self.logger.info("Confirming Email...")
        # Getting the email confirmation link from the email
        self.scrape_emails = True
        while self.scrape_emails:
            emails = tempmail.TempMail.getEmails(self.inbox)
            for mail in emails:
                if "mail.discord.com" in str(mail.sender):
                    for word in mail.body.split():
                        if "https://click.discord.com" in word:
                            self.email_link = word
                            self.scrape_emails = False
                            break
        self.email_link = self.email_link.replace("[", "").replace("]", "")

        self.logger.info("Waiting 10 seconds for a more realistic email-verify")
        await self.page.wait_for_timeout(random.randint(10000, 12000))

        # Confirming the email by link
        await self.page.goto(self.email_link)

        try:
            await self.page.solve_hcaptcha()
        except:
            self.logger.info("No EmailCaptcha detected")

        # Waiting until new token is set
        while self.token == before_token:
            await self.page.wait_for_timeout(1000)

        is_locked = await Discord.is_locked(self)
        if is_locked:
            self.logger.error(f"Token {self.token} got locked whilst verifying the Email!")
            await self.close()
            return

        self.log_output()
        return True



main.py
# PreInstalled PyPackages
import asyncio
import logging
import os
import random
import traceback

# Pip Install Packages
import botright
import httpx
import validators

# Imports from Files
from modules.discord import Discord
from modules.tempmail import TempMail

class Generator:
    async def initialize(self, botright_client, proxy, mode=None, output_file="output.txt", email=True, humanize=True, output_format="token:email:pass", invite_link=""):
        # Initializing the Thread
        self.output_file, self.output_format = output_file, output_format
        self.email_verification, self.humanize, self.invite_link = email, humanize, invite_link
        self.token, self.email, self.output = "", "", ""
        # SettingUp Logger
        logging.basicConfig(
            format='\033[34m[%(levelname)s] - \033[94mLine %(lineno)s  - \033[36m%(funcName)s() - \033[96m%(message)s\033[0m')
        self.logger = logging.getLogger('logger')
        self.logger.setLevel(logging.DEBUG)

        # Initializing Browser and Page
        self.browser = await botright_client.new_browser(proxy)
        self.logger.info("Spawned Browser successfully")

        self.page = await self.browser.new_page()

        if mode == 1:
            await self.generate_unclaimed()
        elif mode == 2:
            await self.generate_token()
        elif mode == 3:
            await self.check_captcha()
        return

    # Helper Functions
    async def log_token(self):
        async def check_json(route, request):
            await route.continue_()
            try:
                response = await request.response()
                await response.finished()
                json = await response.json()
                if json.get("token"):
                    self.token = json.get("token")
            except Exception:
                pass

        await self.page.route("https://discord.com/api/**", check_json)

    def log_output(self):
        output = ""
        for item in self.output_format.split(":"):
            if "token" in item and self.token:
                output += self.token + ":"
            if "email" in item and self.email:
                output += self.email + ":"
            if "pass" in item:
                output += self.browser.faker.password + ":"
            if "proxy" in item and self.browser.proxy:
                output += self.browser.proxy + ":"

        # Remove last :
        output = output[:-1]
        self.output = output

    async def close(self):
        try:
            await self.page.close()
        except:
            pass
        try:
            await self.browser.close()
        except:
            pass

    # Main Functions
    async def generate_unclaimed(self):
        try:
            # Going on Discord Register Site
            try:
                await self.page.goto("https://discord.com/")
            except:
                self.logger.error("Site didn´t load")
                await self.close()
                return False
            # Setting Up TokenLog
            await self.log_token()
            # Click Open InBrowser Button
            await self.page.click('[class *= "gtm-click-class-open-button"]')
            # Typing Username
            await self.page.type('[class *= "username"]', self.browser.faker.username)
            # Clicking Tos and Submit Button
            try:
                await self.page.click("[class *= 'checkbox']", timeout=10000)
            except Exception as e:
                self.logger.debug("No TOS Checkbox was detected")
                pass
            await self.page.click('[class *= "gtm-click-class-register-button"]')

            # Solving Captcha
            await self.page.solve_hcaptcha()

            while not self.token:
                await self.page.wait_for_timeout(2000)

            self.logger.info(f"Generated Token: {self.token}")
            await self.page.wait_for_timeout(2000)

            is_locked = await Discord.is_locked(self)
            if is_locked:
                self.logger.error(f"Token {self.token} is locked!")
                await self.close()
                return
            else:
                self.logger.info(f"Token: {self.token} is unlocked! Flags: {self.flags}")

            self.log_output()

            await self.page.wait_for_timeout(3000)
            try:
                await self.page.type('[id="react-select-2-input"]', self.browser.faker.birth_day)
                await self.page.keyboard.press("Enter")
                await self.page.type('[id="react-select-3-input"]', self.browser.faker.birth_month)
                await self.page.keyboard.press("Enter")
                await self.page.type('[id="react-select-4-input"]', self.browser.faker.birth_year)
                await self.page.keyboard.press("Enter")
                await self.page.wait_for_timeout(1000)
                await self.page.keyboard.press("Enter")
            except:
                pass

            # Closing PopUps
            for _ in range(2):
                try:
                    await self.page.click("[class *= 'closeButton']", timeout=5000)
                except:
                    pass

            if self.email_verification:
                self.inbox = TempMail.generateInbox()
                self.logger.info("Claiming Account...")
                await Discord.set_email(self, self.inbox.address)

                await self.page.wait_for_timeout(2000)

                self.logger.info("Verifying email...")
                await Discord.confirm_email(self)

            self.log_output()

            await self.page.wait_for_timeout(2000)

            if self.humanize:
                await Discord.humanize_token(self)

            await self.page.wait_for_timeout(2000)

            if self.invite_link:
                await Discord.join_server(self)

            self.log_output()
            with open(self.output_file, 'a') as file:
                file.write(f"{self.output}\n")

            self.logger.info("Successfully Generated Account! Closing Browser...")

            await self.close()

        # Catch Exceptions and save output anyways
        except:
            self.logger.error(f"Catched Exception, trying to save Token anyways... \n Error: \n {traceback.format_exc()}")
            if self.output:
                with open(self.output_file, 'a') as file:
                    file.write(f"{self.output}\n")

    async def generate_token(self):
        try:
            # Going on Discord Register Site
            try:
                await self.page.goto("https://discord.com/register")
            except:
                self.logger.error("Site didn´t load")
                await self.close()
                return False
            # Setting Up TokenLog
            await self.log_token()
            # Typing Email, Username, Password
            self.email = f"{self.browser.faker.username}{random.randint(10, 99)}@gmail.com"

            if self.email_verification:
                self.inbox = TempMail.generateInbox()
                self.email = self.inbox.address
            await self.page.type('[name="email"]', self.email)
            await self.page.type('[name="username"]', self.browser.faker.username)
            await self.page.type('[name="password"]', self.browser.faker.password)
            # Typing BirthDay, BirthMonth, BirthYear
            await self.page.type('[id="react-select-2-input"]', self.browser.faker.birth_day)
            await self.page.keyboard.press("Enter")
            await self.page.type('[id="react-select-3-input"]', self.browser.faker.birth_month)
            await self.page.keyboard.press("Enter")
            await self.page.type('[id="react-select-4-input"]', self.browser.faker.birth_year)
            # Clicking Tos and Submit Button
            try:
                tos_box = self.page.locator("[type='checkbox']").first
                await tos_box.click()
            except Exception as e:
                self.logger.debug("No TOS Checkbox was detected")
                pass
            await self.page.click('[type="submit"]')

            await self.page.solve_hcaptcha()

            while not self.token:
                await self.page.wait_for_timeout(2000)

            self.logger.info(f"Generated Token: {self.token}")
            await self.page.wait_for_timeout(2000)

            is_locked = await Discord.is_locked(self)
            if is_locked:
                self.logger.error(f"Token {self.token} is locked!")
                await self.close()
                return
            else:
                self.logger.info(
                    f"Token: {self.token} is unlocked! Flags: {self.flags}")

            self.log_output()

            await self.page.wait_for_timeout(2000)

            # Closing PopUps
            for _ in range(2):
                try:
                    await self.page.click("[class *= 'closeButton']", timeout=5000)
                except:
                    pass

            if self.email_verification:
                self.logger.info("Verifying email...")
                await Discord.confirm_email(self)

            self.log_output()

            await self.page.wait_for_timeout(2000)

            if self.humanize:
                await Discord.humanize_token(self)

            await self.page.wait_for_timeout(2000)

            if self.invite_link:
                await Discord.join_server(self)

            self.log_output()
            with open(self.output_file, 'a') as file:
                file.write(f"{self.output}\n")

            self.logger.info(
                "Successfully Generated Account! Closing Browser...")

            await self.close()

        # Catch Exceptions and save output anyways
        except:
            self.logger.error(f"Catched Exception, trying to save Token anyways... \n Error: \n{traceback.format_exc()}")
            if self.output:
                with open(self.output_file, 'a') as file:
                    file.write(f"{self.output}\n")

    # Testing (Maybe used later?)
    async def login_token(self):
        # Going on Discord Register Site
        try:
            await self.page.goto("https://discord.com/register")
        except:
            self.logger.error("Site didn´t load")
            return False
        await self.page.evaluate(str('setInterval(() => {document.body.appendChild(document.createElement `iframe`).contentWindow.localStorage.token = `"' + self.token + '"`}, 2500); setTimeout(() => {location.reload();}, 2500);'))
        await self.page.wait_for_timeout(5000)

        if self.email_verification:
            self.inbox = TempMail.generateInbox()
            self.logger.info("Claiming Account...")
            await Discord.set_email(self, self.inbox.address)
            await self.page.wait_for_timeout(2000)
            self.logger.info("Verifying email...")
            await Discord.confirm_email(self)
        # self.log_output()
        # await self.page.wait_for_timeout(2000)
        # if self.humanize:
        #     await Discord.humanize_token(self)
        # await self.page.wait_for_timeout(2000)
        if self.invite_link:
            await Discord.join_server(self)
        self.log_output()
        with open(self.output_file, 'a') as file:
            file.write(f"{self.output}\n")
        self.logger.info("Successfully Generated Account! Closing Browser...")
        await self.close()


async def main():
    botright_client = await botright.Botright(headless=False)
    print(""" _____     __     ______     __         ______     ______     __  __
/\  __-.  /\ \   /\  ___\   /\ \       /\  __ \   /\  ___\   /\ \/ /
\ \ \/\ \ \ \ \  \ \___  \  \ \ \____  \ \ \/\ \  \ \ \____  \ \  _"-.
 \ \____-  \ \_\  \/\_____\  \ \_____\  \ \_____\  \ \_____\  \ \_\ \_\\
  \/____/   \/_/   \/_____/   \/_____/   \/_____/   \/_____/   \/_/\/_/    | Made by Vinyzu
                                                                           | https://github.com/Vinyzu/DiscordGenerator""")

    mode = input("[Select] - [Generation Mode]\n" + "<1> Generate Unclaimed Token\n" + "<2> Generate Token\n" + "<3> Test Captcha\n" + "</> ")
    if mode not in ("1", "2", "3"):
        raise ValueError("Invalid Mode provided")
    else:
        mode = int(mode)

    if mode in (1, 2):
        email = input("[Select] - [Email Verification]\n" + "<1> Verification Enabled\n" +
                      "<2> No Verification\n" + "</> ")
        if email not in ("1", "2"):
            raise ValueError("Invalid Mode provided")
        else:
            email = True if email == "1" else False
    else:
        email = False

    if mode in (1, 2):
        humanize = input("[Select] - [Token Humanization]\n" + "<1> Humanization Enabled\n" +
                         "<2> No Humanization\n" + "</> ")
        if humanize not in ("1", "2"):
            raise ValueError("Invalid Mode provided")
        else:
            humanize = True if humanize == "1" else False
    else:
        humanize = False

    threads = input("[Input] - [Threads Amount]\n" + "</> ")
    try:
        threads = int(threads)
    except:
        raise ValueError("Invalid ThreadAmount provided")

    proxy_file = input("[Drag&Drop] - [Proxy File]\n" +
                       "<?> Or Leave empty for Proxyless Mode\n" + "</> ").replace('"', "")
    if proxy_file:
        if not os.path.isfile(proxy_file):
            raise ValueError("Provided ProxyPath isnt a file!")
        proxies = open(proxy_file, 'r').readlines()
    else:
        proxies = None

    output_file = input("[Drag&Drop] - [Output File]\n" +
                        "<?> Or Leave empty to use output.txt\n" + "</> ").replace('"', "")
    if output_file:
        if not os.path.isfile(output_file):
            raise ValueError("Provided OutputPath isnt a file!")
    else:
        output_file = "output.txt"

    invite = input("[Input] - [Invite Link]\n" +
                   "<?> Either parse a InviteLink, or an InviteCode\n" + "</> ")

    if not validators.url(invite) and invite:
        invite_link = f"https://discord.gg/{invite}"
        if not validators.url(invite_link) or not httpx.get(f"https://discordapp.com/api/v8/invites/{invite}").is_success:
            raise ValueError(f"Invalid InviteLink: {invite}")
    else:
        invite_code = invite.split("/")[-1]
        if invite and not httpx.get(f"https://discordapp.com/api/v8/invites/{invite_code}").is_success:
            raise ValueError(f"Invalid InviteLink: {invite}")
        invite_link = invite

    output_format = input("[Input] - [Output Format]\n" +
                          "<?> Token: token, Email: email, Password: pass, Proxy: proxy\n" +
                          "<?> Leave empty for standart output: token:email:pass\n" + "</> ")
    if not output_format:
        output_format = "token:email:pass"
    for item in output_format.split(":"):
        if item not in ["token", "email", "pass", "proxy"]:
            raise ValueError(f"Invalid OutputItem: {item}")

    os.system('cls' if os.name == 'nt' else 'clear')

    try:
        while True:
            threadz = []
            for _ in range(threads):
                proxy = random.choice(proxies) if proxies else None
                threadz.append(Generator().initialize(botright_client, proxy, mode, output_file, email, humanize, output_format, invite_link))

            await asyncio.gather(*threadz)
    except KeyboardInterrupt:
        await botright_client.close()
    except Exception:
        print(traceback.format_exc())
        await botright_client.close()


if __name__ == '__main__':
    asyncio.run(main())
